Description: Reproducible version hash table.
 - For version#"configure arguments", we replace the build directory
   with "/build/path" in all the instances of the -fdebug-prefix-map
   flag.
 - For version#"compile-node-name" and version#"operating system release", we
   use dummy values.
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: not-needed
Last-Update: 2021-09-23

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -6,7 +6,7 @@
 AC_SUBST(CONFIGURED_FILES,"$ac_config_files")
 AC_SUBST(CONFIG_ARGS,"$ac_configure_args")
 echo "'$0' $ac_configure_args" > config.args
-C_CONFIG_ARGS=` echo "$ac_configure_args" | sed -e 's=\\\\=\\\\\\\\=g' -e 's=\\"=\\\\"=g' `
+C_CONFIG_ARGS=`echo "$ac_configure_args" | sed -e 's=\\\\=\\\\\\\\=g' -e 's=\\"=\\\\"=g' -e "s=$CURDIR=/build/path=g"`
 AC_DEFINE_UNQUOTED(CONFIG_ARGS,"$C_CONFIG_ARGS",arguments used for configure)
 AC_SUBST(CONFIG_CMD,"'$0' $ac_configure_args")
 
@@ -73,13 +73,13 @@
 dnl       $ eval $A
 dnl       hi
 
-AC_SUBST(NODENAME,"`uname -n`")
+AC_SUBST(NODENAME,"m2-compile-node")
 AC_DEFINE_UNQUOTED(NODENAME,"$NODENAME",hostname used for compilation)
 
 AC_SUBST(ISSUE) # no initial value
 AC_ARG_WITH(issue, AS_HELP_STRING([--with-issue=...,specify the OS issue (e.g., Ubuntu-7.10)]),ISSUE=$withval)
 AC_SUBST(OS,"`  uname -s | sed s=/=-=g  `")
-AC_SUBST(REL,"`  uname -r | sed -e s=/=-=g -e 's= =+=' -e 's=[()]==g'  `")
+AC_SUBST(REL,"m2-os-release")
 AC_SUBST(UNAME_INFO_COMMAND," uname -pmso 2>/dev/null || uname -pms ")
 AC_SUBST(UNAME_INFO,"` eval $UNAME_INFO_COMMAND `")
 
